neurons/miners/openai/miner.py

```diff
@@ -1,22 +1,156 @@
-# The MIT License (MIT)
-# Copyright © 2024 Yuma Rao
-
-# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
-# documentation files (the “Software”), to deal in the Software without restriction, including without limitation
-# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
-# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
-
-# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
-# the Software.
-
-# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
-# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
-# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-# DEALINGS IN THE SOFTWARE.
 import time
 import bittensor as bt
-from prompting.miners import OpenAIMiner
+import os
+from starlette.types import Send
+from functools import partial
+from typing import Awaitable
+
+# Bittensor Miner Template:
+from prompting.base.prompting_miner import BaseStreamPromptingMiner
+from prompting.base.protocol import StreamPromptingSynapse
+
+# import base miner class which takes care of most of the boilerplate
+
+from neurons.miners.openai.utils import OpenAIUtils
+from dotenv import load_dotenv, find_dotenv
+from openai import OpenAI
+from traceback import print_exception
+
+from prompting import settings
+
+
+class OpenAIMiner(BaseStreamPromptingMiner, OpenAIUtils):
+    """Langchain-based miner which uses OpenAI's API as the LLM.
+    This miner does not use any tools or external APIs when processing requests - it relies entirely on the models' own representation and world model. In some cases, this can produce lower quality results.
+        You should also install the dependencies for this miner, which can be found in the requirements.txt file in this directory.
+    """
+
+    def __init__(self, config=None):
+        super().__init__(config=config)
+
+        bt.logging.info(f"Initializing with model {settings.NEURON_MODEL_ID_MINER}...")
+
+        if settings.WANDB_ON:
+            self.identity_tags = ("openai_miner",) + (settings.NEURON_MODEL_ID_MINER)
+
+        _ = load_dotenv(find_dotenv())
```

Comment:
I just did not notice this - this was already here before I created settings:)

Re find_dotenv() - I generally find it's better to only load .env from the place where it's meant to be rather than dynamically trying to locate it - that quite easily ends up loading e.g. the wrong .env file